gtsam/sfm/tests/testShonanAveraging.cpp

```diff
@@ -91,6 +91,24 @@ TEST(ShonanAveraging3, checkOptimality) {
   EXPECT(!kShonan.checkOptimality(random));
 }
 
+/* ************************************************************************* */
+TEST(ShonanAveraging3, checkSubgraph) {
+  // Create parameter with solver set to SUBGRAPH
+  auto params = ShonanAveragingParameters3(
+      gtsam::LevenbergMarquardtParams::CeresDefaults(), "SUBGRAPH");
+  ShonanAveraging3::Measurements measurements;
+
+  auto subgraphShonan = fromExampleName("toyExample.g2o", params);
```

Comment:
could be helpful to add a small comment explaining what is inside the toyExample.g2o file? i.e. what is it's structure? Is this test designed to make sure a disconnected graph is caught, or instead to operate on just the largest tree in the forest if disconnected?

Reply:
Good advice! The test data is a fully connected graph. Currently Shonan in gtsam doesn't deal with disconnected graph, this is only a simple test case to check whether the subgraph option works. Once I added the part handling the disconnected graph, I will add more test cases.